addons/mail/static/src/models/message_pending/message_pending.js

```diff
@@ -0,0 +1,203 @@
+                body = body.replace(text, placeholder);
+            }
+            for (const channel of this.mentionedChannels) {
+                const placeholder = `#-mention-channel-${channel.id}`;
+                const text = `#${owl.utils.escape(channel.name)}`;
+                mentions.push({
+                    class: 'o_channel_redirect',
+                    id: channel.id,
+                    model: 'mail.channel',
+                    placeholder,
+                    text,
+                });
+                body = body.replace(text, placeholder);
+            }
+            const baseHREF = this.env.session.url('/web');
+            for (const mention of mentions) {
+                const href = `href='${baseHREF}#model=${mention.model}&id=${mention.id}'`;
+                const attClass = `class='${mention.class}'`;
+                const dataOeId = `data-oe-id='${mention.id}'`;
+                const dataOeModel = `data-oe-model='${mention.model}'`;
+                const target = `target='_blank'`;
+                const link = `<a ${href} ${attClass} ${dataOeId} ${dataOeModel} ${target}>${mention.text}</a>`;
+                body = body.replace(mention.placeholder, link);
+            }
+            return body;
+        }
+
+        /**
+         * @private
+         * @param {string} htmlString
+         * @returns {string}
+         */
+        _generateEmojisOnHtml(htmlString) {
+            for (const emoji of emojis) {
+                for (const source of emoji.sources) {
+                    const escapedSource = String(source).replace(
+                        /([.*+?=^!:${}()|[\]/\\])/g,
+                        '\\$1');
+                    const regexp = new RegExp(
+                        '(\\s|^)(' + escapedSource + ')(?=\\s|$)',
+                        'g');
+                    htmlString = htmlString.replace(regexp, '$1' + emoji.unicode);
+                }
+            }
+            return htmlString;
+        }
+
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsLog() {
+            return this.composer.isLog;
+        }
+
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeThread() {
+            return replace(this.composer.thread);
+        }
+
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeRawBody() {
+            return this.composer.textInputContent;
+        }
+
+    }
+
+    PendingMessage.fields = {
+        /**
+         * States which attachments are currently being created in this pending message.
+         */
+        attachments: many2many('mail.attachment', {
+            inverse: 'pendingMessages',
+        }),
+        /**
+         * Determines the body that will be sent to the server.
+         * It will be created by the `generateBody` function when needed to
+         * avoid performance issue
+         */
+        body: attr(),
```

Comment:
Avoid "will" in comment, and also avoid explanation that is too technical and based on current implementation.

Something brief like this is better:
```
/**
 * The body of message resulting from this pending message.
 * It's not set before posting message due to performance reasons.
 */